src/native/fs/fs_napi.cpp

```diff
@@ -619,7 +641,11 @@ struct Stat : public FSWorker
                 GPFS_FCNTL_OR_RETURN(get_fd_gpfs_xattr(fd, _xattr, gpfs_error));
             }
         }
-        CHECK_CTIME_CHANGE(fd, _stat_res, _path);
+
+        // If mtime check is requested instead of ctime check then completely
+        // skip running the check. Change in mtime does not indicate anything other
+        // than that file content.
+        if (!use_mtime_check) CHECK_CTIME_CHANGE(fd, _stat_res, _path);
```

Comment:
TBH this is rather confusing, because we have an option that explicitly say "use mtime" and then we ignore mtime in some cases, so perhaps we should just rename the option to represent the behavior better?